GVFS/GVFS.Service/Handlers/RequestHandler.cs

```diff
@@ -0,0 +1,87 @@
+﻿using GVFS.Common.NamedPipes;
+using GVFS.Common.Tracing;
+using System.Runtime.Serialization;
+
+namespace GVFS.Service.Handlers
+{
+    public class RequestHandler
+    {
+        private string etwArea;
+        private ITracer tracer;
+        private RepoRegistry repoRegistry;
+
+        public RequestHandler(ITracer tracer, string etwArea, RepoRegistry repoRegistry)
+        {
+            this.tracer = tracer;
+            this.etwArea = etwArea;
+            this.repoRegistry = repoRegistry;
+        }
+
+        public void HandleRequest(ITracer tracer, string request, NamedPipeServer.Connection connection)
+        {
+            NamedPipeMessages.Message message = NamedPipeMessages.Message.FromString(request);
+            if (string.IsNullOrWhiteSpace(message.Header))
+            {
+                return;
+            }
+
+            using (ITracer activity = this.tracer.StartActivity(message.Header, EventLevel.Informational, new EventMetadata { { "request", request } }))
+            {
+                string deserializeError = null;
+
+                try
+                {
+                    this.HandleMessage(activity, message, connection, out deserializeError);
+                }
+                catch (SerializationException ex)
+                {
+                    activity.RelatedError("{0}: {1}", deserializeError, ex.Message);
```

Comment:
Error messages should include some context of what was happening (and where) when the error occurred.

Also, include the full exception callstack (preferably with its own key in an `EventMetadata` object)